### PR TITLE
Massive reorg/update based

### PR DESCRIPTION
**Pull request by pablos-here**
Hi,

I used `man awk` sections to reorg/update/copy/create the snippets.  When it made sense, I used the `man` page's section to name the subdirectory.

- I kept the original `contributor` on any snippet that was preexisting.  Whether refreshed or not.  :)
- For new snippets, I didn't bother with a `contributor`.

One slightly annoying item is that in `Emacs`, the sub directories are presented as sub menus but in reverse order.  I'm sure there's a way to solve this issue.  We probably should solve it it and include it in the `README.md`

Speaking of the `README.md`, we might want to update it.  Let me know if you'd like me to take a try.  

Oh, this PR is based on #1. 

Thx!

### CODE REVIEW

File: awk-mode/functions/asorti
Comment (uberkael) [inline awk-mode/functions/asorti:1]: this deleted file is now awk-mode/strings/asorti

Comment (pablos-here) [reply awk-mode/functions/asorti:1]: I should have been clearer.  

No functionality was lost when deleting files.  For example, as you point out above, `asorti`, while deleted, was moved to another directory.  In fact, a directory named `string` as the function `asorti` is in the `man awk` page under `String Functions`.   :)

File: awk-mode/loops/adel
Comment (uberkael) [inline awk-mode/loops/adel:1]: Maybe we can keep this one

Comment (pablos-here) [reply awk-mode/loops/adel:1]: That's been replaced with `control/fora` with a hint on how to access the array element.  One can then elect to use the `del` snippet to delete the array element.

What do you think?

Comment (pablos-here) [reply awk-mode/loops/adel:1]: To make it easier to review, here's the current code for `fora`:

```
for (${1:var} in ${2:array}) {
  # $2[$1]$
  $0
}
```

Comment (uberkael) [reply awk-mode/loops/adel:1]: Sure, but maybe it can be useful to keep it as a utility by itself (even if it is not included in awk).

Comment (pablos-here) [reply awk-mode/loops/adel:1]: I have an idea, how about we create a new sub-directory for utility/template/ready-made snippets?  

If that works for you, I can populate it with `adel`.

Comment (uberkael) [reply awk-mode/loops/adel:1]: Sounds good to me.

Comment (pablos-here) [reply awk-mode/loops/adel:1]: Done!  Please review.  o7

Comment (uberkael) [reply awk-mode/loops/adel:1]: We can include also awk-mode/stream/getcom and awk-mode/stream/getlinevar

**Review by uberkael — APPROVED** [review]

**Comment by pablos-here** [comment]
Hi @uberkael ,

I wasn't quite done ... :) ... please have a look at the updated `README.md`.  I've also added a `LICENSE` that respect the original intention of GPL v3.

Gracias!

**Comment by pablos-here** [comment]
Hey @uberkael , what I'll do is create another PR for the above two changes.  :)  Keep it easy.

**Pull request merged**